src/rdkafka_cgrp.c

```diff
@@ -3285,6 +3321,21 @@ void rd_kafka_cgrp_handle_ConsumerGroupHeartbeat(rd_kafka_t *rk,
                     rkcg, "coordinator query");
         }
 
+        if (actions & RD_KAFKA_ERR_ACTION_SPECIAL &&
+            rkcg->rkcg_flags & RD_KAFKA_CGRP_F_SUBSCRIPTION) {
+                rd_ts_t min_error_interval =
+                    RD_MAX(rkcg->rkcg_heartbeat_intvl_ms * 1000, 3000000);
+                if (rkcg->rkcg_last_err != err ||
+                    (rd_clock() >
+                     rkcg->rkcg_ts_last_err + min_error_interval)) {
+                        rd_kafka_cgrp_set_last_err(rkcg, err);
+                        rd_kafka_consumer_err(
+                            rkcg->rkcg_q, rd_kafka_broker_id(rkb), err, 0, NULL,
+                            NULL, err, "ConsumerGroupHeartbeat failed: %s",
+                            rd_kafka_err2str(err));
```

Comment:
It would be good if we can add the error message as well there. We are extracting `error_str` from the response but not using it anywhere.

Reply:
Tried but the error string is empty for this error, maybe it could be added in future, handled the case where it's empty.